Property resolution fails if you have an env variable that matches the beginning of property name

with version 3.11.2 (with quarkus 3.18.3)

application-dev.properties contains
```
bamboo.myvar=test
```

If I have an environment variable with name bamboo, I get the following error

```
Caused by: java.lang.StringIndexOutOfBoundsException: begin 7, end 6, length 6
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4604)
        at java.base/java.lang.String.substring(String.java:2707)
        at java.base/java.lang.String.substring(String.java:2680)
        at io.smallrye.config.ConfigMappingContext.matchPropertiesWithEnv(ConfigMappingContext.java:243)
        at io.smallrye.config.ConfigMappingContext.<init>(ConfigMappingContext.java:53)
        at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:129)
        at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:126)
        at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:28)
        at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:126)
        at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:93)
        at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:773)
        at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
        at io.quarkus.runtime.generated.Config.createRunTimeConfig(Unknown Source)
        at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
```

--- a/common/src/main/java/io/smallrye/config/common/utils/StringUtil.java
+++ b/common/src/main/java/io/smallrye/config/common/utils/StringUtil.java
@@ -199,7 +199,7 @@ public class StringUtil {
      */
     public static boolean isInPath(final String path, final String name) {
         if (name.equals(path)) {
-            return true;
+            return false;
         }
 
         // if property is less than the root no way to match

--- a/implementation/src/test/java/io/smallrye/config/EnvConfigSourceTest.java
+++ b/implementation/src/test/java/io/smallrye/config/EnvConfigSourceTest.java
@@ -672,6 +672,19 @@ class EnvConfigSourceTest {
         }
     }
 
+    @Test
+    void prefixAsEnv() {
+        SmallRyeConfig config = new SmallRyeConfigBuilder()
+                .withSources(new EnvConfigSource(Map.of("PREFIX", "VALUE", "PREFIX_VALUE", "VALUE"), 300))
+                .withMapping(PrefixAsEnv.class)
+                .build();
+    }
+
+    @ConfigMapping(prefix = "prefix")
+    interface PrefixAsEnv {
+        Optional<String> value();
+    }
+
     private static boolean envSourceEquals(String name, String lookup) {
         return BOOLEAN_CONVERTER.convert(new EnvConfigSource(Map.of(name, "true"), 100).getValue(lookup));
     }

